File: Python/0890_find_and_replace_pattern/solution.py
def find_and_replace_pattern(words,pattern):
    result = list()
    for word in words:
        check_dict = dict()
        for index in range(len(pattern)):
            if pattern[index] not in check_dict.keys() and word[index] not in check_dict.values():
                check_dict[pattern[index]] = word[index]
        string = str()
        for letter in pattern:
            if letter in check_dict.keys():
                string += check_dict[letter]
        if string == word:
            result.append(string)
    return result

words = ["abc","deq","mee","aqq","dkd","ccc"]
pattern = "abb"
print(find_and_replace_pattern(words,pattern))

File: JavaScript/0890_find_and_replace_pattern/solution.js
function find_and_replace_pattern(words,pattern) {
    let result = [];
    for (let word of words) {
        let check_dict = {};
        for (let index = 0; index < pattern.length; index ++) {
            if (!Object.keys(check_dict).includes(pattern[index]) && !Object.values(check_dict).includes(word[index])) {
                check_dict[pattern[index]] = word[index];
            }
        }
        let string = "";
        for (let letter of pattern) {
            if (Object.keys(check_dict).includes(letter)) {
                string += check_dict[letter];
            }
        }
        if (string === word) {
            result.push(string);
        }
    }
    return result;
}

const words = ["abc","deq","mee","aqq","dkd","ccc"];
const pattern = "abb";
console.log(find_and_replace_pattern(words,pattern));